code-challenges.test.js

```diff
@@ -14,39 +20,123 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const hitchhikersCharacters = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+// describe('capKey',()=>{
+// const hitchhikersCharacters = [
+//   { name: "ford prefect", occupation: "a hitchhiker" },
+//   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+//   { name: "arthur dent", occupation: "a radio employee" }
+// ]
+// // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+//   it('returns an array with a sentence about each person with their name capitalized', () => {
+//     expect(capKey(hitchhikersCharacters)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+//   })
+// })
+
+
+
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// Name: capKey
+// Input: Array
+// Output: Array
+// Process: Create a function that takes in an array and returns an array with a sentence about each person with their name capitalized by using string interpolation, charAt() and toUpperCase
+
+const capKey = (key, key1) => {
+ const newArr = []
+ for(let i = 0; i < key.length; i++){
+  const person = key[i]
+  const job = key1[i]
+  const capName = person.name.charAt(0).toUpperCase() + person.name.slice(1)
+   newArr = `${capName} is a ${job} `
```

Comment:
Close! Most of the logic is there. The function should only take 1 parameter, since the argument we are passing in the test is just 1 array. From there, you need to access each object, and then access each name in each object. Then you need to work with that string to capitalize each first letter. Make sure to pay attention during assessment review on this one.